src/linker/Linker.Dataflow/MethodBodyScanner.cs

```diff
@@ -900,15 +967,19 @@ private static VariableDefinition GetLocalDef (Instruction operation, Collection
 			SingleValue? newObjValue;
 			ValueNodeList methodParams = PopCallArguments (currentStack, calledMethod, callingMethodBody, isNewObj,
 														   operation.Offset, out newObjValue);
-
+			ValueNodeList dereferencedMethodParams = new (methodParams.Select (param => DereferenceValue (param, locals)).ToList ());
 			MultiValue methodReturnValue;
 			bool handledFunction = HandleCall (
 				callingMethodBody,
 				calledMethod,
 				operation,
-				methodParams,
+				dereferencedMethodParams,
 				out methodReturnValue);
 
+			// Handle ref params
+			ValueNodeList methodParametersNoThis = calledMethod.HasImplicitThis () ? new ValueNodeList (methodParams.Skip (1).ToList ()) : methodParams;
+			HandleRefParameters (callingMethodBody, calledMethod, methodParametersNoThis, operation, currentStack, locals, curBasicBlock);
```

Comment:
Personally I would prefer if we did this below near the foreach which handles arrays.
And I would add a comment that this is about handling the "out" part of the parameters (which the arrays effectively do as well).
I would either inline the method here, or rename it to make it clear that this handling the "out" part of the ref parameters - the "in" part is handled elsewhere (basically by the HandleCall).

In either case I would get rid of the `methodParametersNoThis` and make the loop handle this (typically we compute an `offset` which we add to the index from the loop when accessing the argument array).